fluxsvc panic with nil os.FileInfo on prod

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x73b922]

goroutine 99 [running]:
panic(0x16b26a0, 0xc420016030)
    /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/weaveworks/flux/platform/kubernetes.FindDefinedServices.func1(0xc420b51d10, 0x24, 0x0, 0x0, 0x222dd20, 0xc420b51dd0, 0x187e101, 0xc420929090)
    /home/ubuntu/src/github.com/weaveworks/flux/platform/kubernetes/files.go:26 +0x32
path/filepath.Walk(0xc420b51d10, 0x24, 0xc420929090, 0x1a, 0x0)
    /usr/local/go/src/path/filepath/path.go:396 +0x8b
github.com/weaveworks/flux/platform/kubernetes.FindDefinedServices(0xc420b51d10, 0x24, 0x24, 0xc420c661c0, 0x38)
    /home/ubuntu/src/github.com/weaveworks/flux/platform/kubernetes/files.go:33 +0x11c
github.com/weaveworks/flux/release.(*ReleaseContext).FindDefinedServices(0xc420ae3260, 0xc420c661c0, 0x38, 0xc420929052, 0x3, 0xc420929055)
    /home/ubuntu/src/github.com/weaveworks/flux/release/context.go:166 +0x5b
github.com/weaveworks/flux/release.(*ReleaseContext).SelectServices(0xc420ae3260, 0xc420929070, 0x1, 0x1, 0xc420b51c80, 0xc420b51cb0, 0xc420b51b00, 0xc420ae37d8, 0xcab, 0x0, ...)
    /home/ubuntu/src/github.com/weaveworks/flux/release/context.go:89 +0x61
github.com/weaveworks/flux/release.selectServices(0xc420ae3260, 0xc420ae3358, 0xc420b51b00, 0xc420ae37d8, 0x0, 0x0, 0x22b5f80, 0xc42052d110, 0x0)
    /home/ubuntu/src/github.com/weaveworks/flux/release/releaser.go:282 +0x4ba
github.com/weaveworks/flux/release.(*Releaser).release(0xc4201c7500, 0xc420928088, 0x4, 0xc420de37c0, 0xc420ae37d8, 0xc420ae37b8, 0x0, 0x0, 0x0, 0x0, ...)
    /home/ubuntu/src/github.com/weaveworks/flux/release/releaser.go:109 +0x607
github.com/weaveworks/flux/release.(*Releaser).Handle(0xc4201c7500, 0xc420de37c0, 0x7fb4f3046050, 0xc4201c7300, 0xc4201c7300, 0xc420928001, 0x7, 0xc4209280e7, 0x7)
    /home/ubuntu/src/github.com/weaveworks/flux/release/releaser.go:68 +0xe8
github.com/weaveworks/flux/jobs.(*Worker).Work(0xc4203dd220)
    /home/ubuntu/src/github.com/weaveworks/flux/jobs/worker.go:94 +0x11ba
created by main.main
    /home/ubuntu/src/github.com/weaveworks/flux/cmd/fluxsvc/main.go:216 +0x21ed
```

--- a/platform/kubernetes/files.go
+++ b/platform/kubernetes/files.go
@@ -22,7 +22,10 @@ func FindDefinedServices(path string) (map[flux.ServiceID][]string, error) {
 	}
 
 	var files []string
-	filepath.Walk(path, func(target string, info os.FileInfo, err error) error {
+	if err = filepath.Walk(path, func(target string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
 		if info.IsDir() {
 			return nil
 		}
@@ -30,7 +33,9 @@ func FindDefinedServices(path string) (map[flux.ServiceID][]string, error) {
 			files = append(files, target)
 		}
 		return nil
-	})
+	}); err != nil {
+		return nil, err
+	}
 
 	services := map[flux.ServiceID][]string{}
 	for _, file := range files {

